paramiko/transport.py

```diff
@@ -67,20 +67,990 @@
                                     ChannelException, ProxyCommandFailure)
 from paramiko.util import retry_on_signal, ClosingContextManager, clamp_value
 
+# for thread cleanup
+_active_threads = []
+
+def _join_lingering_threads():
+    for thr in _active_threads:
+        thr.stop_thread()
+
+import atexit
+atexit.register(_join_lingering_threads)
+
+
+class Transport(object):
+    """
+    An SSH Transport attaches to a stream (usually a socket), negotiates an
+    encrypted session, authenticates, and then creates stream tunnels, called
+    `channels <.Channel>`, across the session.  Multiple channels can be
+    multiplexed across a single session (and often are, in the case of port
+    forwardings).
+
+    Instances of this class may be used as context managers.
```

Comment:
For this comment to be true, this `Transport` class should sub-class `ClosingContextManager` (and `TransportThread` probably no longer needs to sub-class it).